release/sil/sil.km.ggc/source/sil.km.ggc.model.ts

```diff
@@ -10,59 +10,71 @@
 const source: LexicalModelSource = {
   format: 'trie-1.0',
   sources: ['wordlist.tsv'],
-        wordBreaker: function (str) {
-  const tokens = str.split(/\s|\u200b/);
+  wordBreaker: function (str) {
+    const whitespaceRegex = /\s|\u200b|\n|\r/;
+    const plainTokens = str.split(whitespaceRegex);
 
-  for(let i=0; i < tokens.length; i++) {
-    const token = tokens[i];
-    if(token.length == 1) {
-      continue;
-    }
+    for(let i=0; i < plainTokens.length; i++) {
+      const token = plainTokens[i];
+      if(token.length == 0) {
+        plainTokens.splice(i, 1);
+        i--;
+        continue;
+      } else if(token.length == 1 && whitespaceRegex.test(token)) {
+        plainTokens.splice(i, 1);
+        i--;
+        continue;
+      }
```

Comment:
Prevents empty tokens from appearing in the middle of context, especially should excessive whitespaces exist between words.  Something like `the        quick   brown fox` should appear to skip all spaces between tokens at once, not in multiple steps.